posthog/async_migrations/migrations/0002_events_sample_by.py

```diff
@@ -30,6 +32,25 @@
 """
 
 
+def generate_insert_into_op(partition_gte: int, partition_lt=None) -> AsyncMigrationOperation:
+    lt_expression = f"toYYYYMM(timestamp) < {partition_lt} AND" if partition_lt else ""
+    op = AsyncMigrationOperation(
+        database=AnalyticsDBMS.CLICKHOUSE,
+        sql=f"""
+        INSERT INTO {TEMPORARY_TABLE_NAME}
+        SELECT *
+        FROM {EVENTS_TABLE}
+        WHERE 
+            toYYYYMM(timestamp) >= {partition_gte} AND {lt_expression}
+            timestamp >= (SELECT max(timestamp) FROM {TEMPORARY_TABLE_NAME})
```

Comment:
What's the goal with this? Is it to make sure we don't double-insert?

If so, I don't think it works since we don't enforce ordering in this insert - I don't think there's a performant way to enforce it either.

A better way might be to do an `OPTIMIZE TABLE` at the end to deduplicate.

Reply:
goal: yes when resuming. 👍  for pointing out the problem and solution.

Reply:
good point. we do already do an `OPTIMIZE TABLE` in this migration, so it makes sense to just drop this check and embrace the duplicates from a potential resuming